core/src/actors/paths.rs

```diff
@@ -189,6 +189,11 @@ impl SystemPath {
         self.into_named_with_vec(parsed)
     }
 
+    /// Returns the SocketAddr corresponding to the SystemPath
+    pub(crate) fn socket_address(&self) -> SocketAddr {
```

Comment:
I'm actually not too opposed to having this as a public API, too. It could help interact with external code in some places, I think.